src/ensembl/production/metadata/api/genome.py

```diff
@@ -249,48 +398,106 @@ def fetch_sequences_by_assembly_accession(
+            Each tuple contains the following elements:
+                - Genome: An instance of the Genome class.
+                - Organism: An instance of the Organism class.
+                - GenomeDataset: An instance of the GenomeDataset class.
+                - Dataset: An instance of the Dataset class.
+                - DatasetType: An instance of the DatasetType class.
+                - DatasetSource: An instance of the DatasetSource class.
+                - EnsemblRelease: An instance of the EnsemblRelease class.
+                - DatasetAttribute: An instance of the DatasetAttribute class.
+                - Attribute: An instance of the Attribute class.
+
+        Raises:
+            ValueError: If an exception occurs during the fetch process.
+
+        """
         try:
             genome_select = db.select(
                 Genome,
+                Organism,
                 GenomeDataset,
                 Dataset,
                 DatasetType,
-                DatasetSource
+                DatasetSource,
             ).select_from(Genome) \
+                .join(Organism, Organism.organism_id == Genome.organism_id) \
                 .join(GenomeDataset, Genome.genome_id == GenomeDataset.genome_id) \
                 .join(Dataset, GenomeDataset.dataset_id == Dataset.dataset_id) \
                 .join(DatasetType, Dataset.dataset_type_id == DatasetType.dataset_type_id) \
-                .join(DatasetSource, Dataset.dataset_source_id == DatasetSource.dataset_source_id)
+                .join(DatasetSource, Dataset.dataset_source_id == DatasetSource.dataset_source_id) \
 
             # set default group topic as 'assembly' to fetch unique datasource
             if not dataset_name:
                 dataset_name = "assembly"
 
             genome_id = check_parameter(genome_id)
             genome_uuid = check_parameter(genome_uuid)
+            organism_uuid = check_parameter(organism_uuid)
             dataset_uuid = check_parameter(dataset_uuid)
             dataset_name = check_parameter(dataset_name)
             dataset_source = check_parameter(dataset_source)
+            dataset_type = check_parameter(dataset_type)
 
             if genome_id is not None:
                 genome_select = genome_select.filter(Genome.genome_id.in_(genome_id))
 
             if genome_uuid is not None:
                 genome_select = genome_select.filter(Genome.genome_uuid.in_(genome_uuid))
 
+            if organism_uuid is not None:
+                genome_select = genome_select.filter(Organism.organism_uuid.in_(organism_uuid))
+
             if dataset_uuid is not None:
                 genome_select = genome_select.filter(Dataset.dataset_uuid.in_(dataset_uuid))
 
             if unreleased_datasets:
                 genome_select = genome_select.filter(GenomeDataset.release_id.is_(None)) \
                     .filter(GenomeDataset.is_current == 0)
```

Comment:
Don't Filter off of is_current for unreleased. Old datasets will be present. Filter off the absence of a release.

Reply:
fixed